autopilot/contractor/hostscore.go

```diff
@@ -61,33 +60,49 @@ func hostScore(cfg api.AutopilotConfig, h api.Host, expectedRedundancy float64)
 //   - If the host is more expensive than expected, an exponential malus is applied.
 //     A 2x ratio will already cause the score to drop to 0.16 and a 3x ratio causes
 //     it to drop to 0.05.
-func priceAdjustmentScore(hostCostPerPeriod types.Currency, cfg api.ContractsConfig) float64 {
-	// return early if the allowance or amount of hosts is zero, avoiding a
-	// division by zero panic below.
-	if cfg.Allowance.IsZero() || cfg.Amount == 0 {
+func priceAdjustmentScore(pt rhpv3.HostPriceTable, gs api.GougingSettings) float64 {
+	// combine the upload and download prices to get a threshold
+	dppb, overflow := gouging.DownloadPricePerByte(pt)
+	if overflow {
 		return math.SmallestNonzeroFloat64
 	}
+	uppb, overflow := gouging.UploadPricePerByte(pt)
+	if overflow {
+		return math.SmallestNonzeroFloat64
+	}
+	sppb := pt.WriteStoreCost
 
-	hostPeriodBudget := cfg.Allowance.Div64(cfg.Amount)
+	priceScore := func(actual, max types.Currency) float64 {
+		threshold := max.Div64(2)
+		if threshold.IsZero() {
+			return 1.0 // no gouging settings defined
+		}
 
-	ratio := new(big.Rat).SetFrac(hostCostPerPeriod.Big(), hostPeriodBudget.Big())
-	fRatio, _ := ratio.Float64()
-	switch ratio.Cmp(new(big.Rat).SetUint64(1)) {
-	case 0:
-		return 0.5 // ratio is exactly 1 -> score is 0.5
-	case 1:
-		// cost is greater than budget -> score is in range (0; 0.5)
-		//
-		return 1.5 / math.Pow(3, fRatio)
-	case -1:
-		// cost < budget -> score is (0.5; 1]
-		s := 0.44 + 0.06*(1/fRatio)
-		if s > 1.0 {
-			s = 1.0
+		ratio := new(big.Rat).SetFrac(actual.Big(), threshold.Big())
+		fRatio, _ := ratio.Float64()
+		switch ratio.Cmp(new(big.Rat).SetUint64(1)) {
+		case 0:
+			return 0.5 // ratio is exactly 1 -> score is 0.5
+		case 1:
+			// actual is greater than threshold -> score is in range (0; 0.5)
+			//
+			return 1.5 / math.Pow(3, fRatio)
+		case -1:
+			// actual < threshold -> score is (0.5; 1]
+			s := 0.5 * (1 / fRatio)
+			if s > 1.0 {
+				s = 1.0
+			}
+			return s
 		}
-		return s
+		panic("unreachable")
 	}
-	panic("unreachable")
+
+	// compute scores for download, upload and storage and combine them
+	downloadScore := priceScore(dppb, gs.MaxDownloadPrice)
+	uploadScore := priceScore(uppb, gs.MaxUploadPrice)
+	storageScore := priceScore(sppb, gs.MaxStoragePrice)
+	return (downloadScore + uploadScore + storageScore) / 3.0
```

Comment:
@n8maninger assigned you as well since this PR is the change we talked about in NY. 

You proposed using the gouging settings for price scoring and this is what I came up with. Similar to what it was before but using gouging settings and splitting up the score into 3 parts for uploads, downloads and storage which are combined into one.